front/displaypreference.form.php

```diff
@@ -66,7 +66,7 @@
 
 // Datas may come from GET or POST : use REQUEST
 if (isset($_REQUEST["itemtype"])) {
-    $setupdisplay->display(['displaytype' => $_REQUEST['itemtype']]);
+    $setupdisplay->display(['displaytype' => addslashes($_REQUEST['itemtype'])]);
```

Comment:
Instead of this localized change, I propose following change:

```diff
diff --git a/src/CommonGLPI.php b/src/CommonGLPI.php
index 064b67f2a4..072b504e8a 100644
--- a/src/CommonGLPI.php
+++ b/src/CommonGLPI.php
@@ -874,7 +874,7 @@ JAVASCRIPT;
             }
 
            // prevent double sanitize, because the includes.php sanitize all data
-            $cleaned_options = Toolbox::stripslashes_deep($cleaned_options);
+            $cleaned_options = Sanitizer::unsanitize($cleaned_options);
 
             $extraparamhtml = "&amp;" . Toolbox::append_params($cleaned_options, '&amp;');
         }
```

`Toolbox::stripslashes_deep()` is doing more than just reverting sanitization process, and may alter data.